File: hcx-pipeline-jobs/coverage-eligibility-check-job/src/main/scala/org/swasth/dp/coverageeligiblitycheck/functions/CoverageEligibilityProcessFunction.scala
package org.swasth.dp.coverageeligiblitycheck.functions

import org.apache.flink.api.common.typeinfo.TypeInformation
import org.slf4j.LoggerFactory
import org.swasth.dp.core.function.{BaseDispatcherFunction, ValidationResult}
import org.swasth.dp.coverageeligiblitycheck.task.CoverageEligibilityCheckConfig

import java.util

class CoverageEligibilityProcessFunction(config: CoverageEligibilityCheckConfig)(implicit val stringTypeInfo: TypeInformation[String])
  extends BaseDispatcherFunction(config) {

  private[this] val logger = LoggerFactory.getLogger(classOf[CoverageEligibilityProcessFunction])

  override def validate(event: util.Map[String, AnyRef]): ValidationResult = {
    // TODO: Add domain specific validations
    ValidationResult(true, None)
  }

}

File: hcx-pipeline-jobs/coverage-eligibility-job/src/main/java/org/swasth/dp/coverageeligiblity/functions/CoverageEligibilityProcessFunction.java
package org.swasth.dp.coverageeligiblity.functions;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.swasth.dp.core.function.BaseDispatcherFunction;
import org.swasth.dp.core.function.ValidationResult;
import org.swasth.dp.coverageeligiblity.task.CoverageEligibilityConfig;

import java.util.Map;

public class CoverageEligibilityProcessFunction extends BaseDispatcherFunction {

    private Logger logger = LoggerFactory.getLogger(CoverageEligibilityProcessFunction.class);
    private CoverageEligibilityConfig config;

    public CoverageEligibilityProcessFunction(CoverageEligibilityConfig config) {
        super(config);
        this.config = config;
    }

    @Override
    public ValidationResult validate(Map<String, Object> event) {
        // TODO: Add domain specific validations
        return new ValidationResult(true, null);
    }

}
